core/tests/puppeteer-acceptance-tests/user-utilities/logged-in-users-utils.ts

```diff
@@ -704,6 +741,278 @@ export class LoggedInUser extends BaseUser {
     }
   }
 
+  /**
+   * Function for navigating to the creator dashboard page.
+   */
+  async navigateToCreatorDashboardPage(): Promise<void> {
+    await this.goto(creatorDashboardUrl);
+  }
+
+  /**
+   * Function for navigating to the learner dashboard page.
+   */
+  async navigateToLearnerDashboardPage(): Promise<void> {
+    await this.goto(learnerDashboardUrl);
+  }
+
+  /**
+   * Function for navigating to the learner dashboard page.
+   */
+  async openCommunityLessonsTab(): Promise<void> {
+    if (this.page.url() !== learnerDashboardUrl) {
+      await this.goto(learnerDashboardUrl);
+    }
+    await this.page.click(communityLessonsTabButton);
+    await this.page.waitForSelector('h1.lessons-title');
+  }
+
+  /**
+   * Function for navigating to the creator dashboard page.
+   */
+  async navigateToCommunityLibraryPage(): Promise<void> {
+    await this.goto(communityLibraryUrl);
+  }
+
+  /**
+   * Function for opening an exploration from the community library page.
+   */
+  async searchAndOpenExplorationFromCommunityLibrary(
+    explorationName: string
+  ): Promise<void> {
+    if (this.page.url() !== communityLibraryUrl) {
+      await this.navigateToCommunityLibraryPage();
+    }
+    await this.findExplorationInCommunityLibrary(explorationName);
+    await this.page.click(communityExplorationCard);
+    await this.page.waitForNavigation();
+  }
+
+  /**
+   * This function checks whether we have completed an exploration or not.
+   */
+  async expectReachedTheEndOfExploration(): Promise<void> {
+    if (await this.page.waitForSelector(explorationCompletionToast)) {
+      showMessage('Exploration completed successfully!');
+    } else {
+      throw new Error('Failed to complete the exploration.');
+    }
+  }
+
+  /**
+   * Function for reporting an exploration.
+   */
+  async reportTheExploration(reportMessage: string): Promise<void> {
+    await this.page.click(reportExplorationButton);
+    await this.page.waitForSelector('.modal-content');
+    await this.page.click(reportOtherOptionSelector);
+    await this.page.waitForSelector(reportTextArea);
+    await this.type(reportTextArea, reportMessage);
+    await this.page.click('.e2e-test-submit-report-button');
+    await this.page.waitForSelector('.modal-content');
+    await this.page.click(
+      'oppia-exploration-successfully-flagged-modal button'
+    );
+    await this.page.waitForSelector(explorationFeedbackButton);
+    showMessage(`Exploration reported with message ${reportMessage}.`);
+  }
+
+  /**
+   * Function for giving feedback for an exploration.
+   */
+  async giveFeedbackToExploration(feedbackMessage: string): Promise<void> {
+    await this.clickOn(explorationFeedbackButton);
```

Comment:
`await this.clickOn(explorationFeedbackButton)` calls `this.page.waitForSelector(explorationFeedbackButton)` before calling `this.page.click(explorationFeedbackButton)`, so the statement on line 815 should be removed due to it being repeated code.

Reply:
Thanks. removed it